### PR TITLE
Album detail and Photo detail pages

### LINKED ISSUE #5: Style album detail

### PR DESCRIPTION
**Pull request by natalieacevedo**
Created photo detail and album detail pages.

Fixes #5 and #6.

### CODE REVIEW

File: src/App.scss [hunk 1]
```diff
@@ -1,4 +1,51 @@
...
+
+  box-sizing: border-box;
+}
+.subtitle {
+  text-align: center;
+  margin-top: 2rem;
+  margin-bottom: 1rem;
+}
+@media only screen and (max-width: 500px) {
+  // .Main {
```
Comment (haplo) [inline src/App.scss]: Avoid committing commented out code.

File: src/App.scss [hunk 2]
```diff
@@ -1,4 +1,51 @@
...
+  box-sizing: border-box;
+}
+.container-fluid {
+  margin: 0;
+  padding: 0;
+}
+.routes-container {
+  height: 100%;
+}
+/////////////////por ahora
```
Comment (haplo) [inline src/App.scss]: Always in English.

File: src/App.scss [hunk 3]
```diff
@@ -1,4 +1,51 @@
...
+*:after {
   margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+.container-fluid {
+  margin: 0;
+  padding: 0;
+}
+.routes-container {
```
Comment (haplo) [inline src/App.scss:20]: Would be nice to have some sorting logic in the CSS files, e.g. alphabetical.

File: src/assets/Rightarrow.tsx
Comment (haplo) [inline src/assets/Rightarrow.tsx:1]: File should be *RightArrow.tsx*.

File: src/assets/Rightarrow.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable react/no-unknown-property */
+import React from "react"
+function RigthArrow() {
```
Comment (haplo) [inline src/assets/Rightarrow.tsx]: Typo, `RightArrow`.

File: src/assets/Rightarrow.tsx
Comment (haplo) [inline src/assets/Rightarrow.tsx:1]: File should be _RightArrow.tsx_, with a capital `A`.

File: src/assets/x.svg [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-x" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
```
Comment (haplo) [inline src/assets/x.svg]: Make it a component like the rest.

File: src/assets/hamburguer.svg [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-menu-2" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
```
Comment (haplo) [inline src/assets/hamburguer.svg]: Make it a component like the rest.

File: src/components/Drawer.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    return () => {
+      window.removeEventListener("resize", isNavShowing)
+    }
+  }, [windWidth])
+
+  return (
+    <div
+      className={`drawer-container ${
+        isOpen || windWidth > 900 ? "drawer-is-open-container" : ""
```
Comment (haplo) [inline src/components/Drawer.tsx]: Avoid magic numbers, move `900` to a module-level constant, e.g. `DESKTOP_WIDTH_THRESHOLD`.

File: src/components/Footer.tsx [hunk 8]
```diff
@@ -1,14 +1,13 @@
 import React from "react"
-import { Container, Row, Col } from "react-bootstrap"
 
 const Footer = () => (
-  <Container>
-    <Row>
-      <Col className="border-top text-center pt-2 mt-3">
+  <div>
```
Comment (haplo) [inline src/components/Footer.tsx]: Are three nested divs necessary?

File: src/components/Header.tsx [hunk 9]
```diff
@@ -1,38 +1,32 @@
 import React from "react"
-import { Button, Container, Form, FormControl, Nav, Navbar } from "react-bootstrap"
 
+import logoAzul from "../assets/clogo.png"
```
Comment (haplo) [inline src/components/Header.tsx]: English.

File: src/components/Header.tsx [hunk 10]
```diff
@@ -1,38 +1,32 @@
...
+
+//import { Button, Container, Form, FormControl, Nav, Navbar } from "react-bootstrap";
```
Comment (haplo) [inline src/components/Header.tsx]: Remove commented out code.

File: src/components/NavigationDrawer.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import BreadCrumbs from "./BreadCrumbs"
+
+type DrawerProps = {
+  photo: PhotoData
+  toggleDrawer: boolean
+  albumId: string
+}
+
+function NavigationDrawer({ photo, toggleDrawer, albumId }: DrawerProps) {
+  console.log(photo.image_url)
```
Comment (haplo) [inline src/components/NavigationDrawer.tsx]: Remove log.

File: src/components/Photo.tsx [hunk 12]
```diff
@@ -1,30 +1,100 @@
...
+
+        <Drawer photo={photo} />
+        <NavigationDrawer photo={photo} toggleDrawer={isOpen} albumId={albumId} />
+      </div>
+    )
+  }
+
+  if (photo && !albumId) {
+    const { id } = photo
+    const realPhotoLink = `${baseUrl}/api/v1/photo/${id}/file`
```
Comment (haplo) [inline src/components/Photo.tsx:87]: This should be moved to the _data_ module. There should be no API URLs in components or pages.

File: src/components/PhotoList.tsx [hunk 13]
```diff
@@ -1,9 +1,9 @@
 import React from "react"
-import { Container, Row, Spinner } from "react-bootstrap"
 import { PhotoListItem, APIError } from "../data/types"
-
-import PhotoThumb from "./PhotoThumb"
-
+import getRandomPic from "../data/apiFoto"
+import PhotoAlbum from "react-photo-album"
```
Comment (haplo) [inline src/components/PhotoList.tsx:2]: Keep third-party imports above app imports.

File: src/components/PhotoThumb.tsx [hunk 14]
```diff
@@ -1,26 +1,38 @@
 import React from "react"
-import { Card, Col } from "react-bootstrap"
 import { PreviousOrNext } from "../data/types"
-
 import { Link } from "react-router-dom"
-
-interface PhotoThumbParams {
-  photo: PreviousOrNext
+interface NavigationDrawerParams {
```
Comment (haplo) [inline src/components/PhotoThumb.tsx]: Should be `PhotoThumbParams`.

File: src/context/drawersContext.tsx
Comment (haplo) [inline src/context/drawersContext.tsx:1]: Let's call it _Drawer.tsx_, not necessary to have _Context_ in the name as it's already in the _contexts_ module.

BTW change _context_ to plural, _contexts_, to match the other modules _pages_ and _components_.

File: src/context/drawersContext.tsx [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useState } from "react"
+
+//Drawer context
```
Comment (haplo) [inline src/context/drawersContext.tsx]: This comment doesn't help, either improve it to explain what the drawer is about or remove it.

File: src/data/apiFoto.ts
Comment (haplo) [inline src/data/apiFoto.ts:1]: Use always English, rename to `apiPhoto`.

File: src/pages/PhotoInAlbumPage.tsx [hunk 16]
```diff
@@ -1,23 +1,47 @@
 import React from "react"
-
-import { Image, Spinner } from "react-bootstrap"
+import { useEffect, useState } from "react"
 import { Link } from "react-router-dom"
 import Photo from "../components/Photo"
-import PhotoThumb from "../components/PhotoThumb"
 import { useParams } from "react-router-dom"
```
Comment (haplo) [inline src/pages/PhotoInAlbumPage.tsx]: Keep third-party imports at the top of the file, followed by app imports.

File: src/pages/PhotoInAlbumPage.tsx [hunk 17]
```diff
@@ -1,23 +1,47 @@
...
+      toggleHeader(true)
+    }
+  }, [toggleHeader, windWidth])
+
   const { albumId, photoId } = useParams()
   const { photoInAlbum } = useApi()
   if (!albumId || !photoId) {
     console.warn("empty props in PhotoInAlbumPage")
     return null
+    console.warn("empty props in PhotoInAlbumPage")
```
Comment (haplo) [inline src/pages/PhotoInAlbumPage.tsx]: Duplicated line due to mishandled merge conflict.

File: src/pages/PhotoInAlbumPage.tsx [hunk 18]
```diff
@@ -1,23 +1,47 @@
...
+    return null
   }
   const { data, error, isError, isLoading } = photoInAlbum({ albumId, photoId })
 
   if (isLoading) {
     return <h1>Info is loading</h1>
+    return <h1>Info is loading</h1>
```
Comment (haplo) [inline src/pages/PhotoInAlbumPage.tsx]: Duplicated line due to mishandled merge conflict.

File: src/pages/PhotoPage.tsx [hunk 19]
```diff
@@ -1,19 +1,28 @@
 import React from "react"
-import { Image, Spinner } from "react-bootstrap"
+
 import Photo from "../components/Photo"
-import PhotoThumb from "../components/PhotoThumb"
+import { useEffect } from "react"
```
Comment (haplo) [inline src/pages/PhotoPage.tsx]: Keep third-party imports at the top of the file, followed by app imports.

File: src/styles/arrowsStyles.scss [hunk 20]
```diff
@@ -0,0 +1,58 @@
+button {
+  -webkit-appearance: none;
+  background: transparent;
+  border: 0;
+  outline: 0;
+}
+
+svg {
```
Comment (haplo) [inline src/styles/arrowsStyles.scss]: This is a global style, should probably go in main file.

File: src/styles/arrowsStyles.scss [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+.svg-arrow {
+  cursor: pointer;
+  position: absolute;
+  top: 50%;
+  margin-top: -45px;
+  margin-left: -35px;
+  width: 70px;
+  height: 90px;
+}
+.up {
```
Comment (haplo) [inline src/styles/arrowsStyles.scss]: These class names are too generic, I suggest you make them `.svg-arrow .up`.

File: src/styles/breadCrumbs.scss [hunk 22]
```diff
@@ -0,0 +1,199 @@
+@use "./mixins.scss" as mix;
+.bread-crumb-link,
```
Comment (haplo) [inline src/styles/breadCrumbs.scss]: Use `breadcrumb` without the hyphen for consistency with other classes.

File: src/styles/mixins.scss [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+  border: 2px solid $secondary-color;
+  border-radius: 10px;
+  max-width: $max-width;
+  color: black;
+  text-align: center;
+  width: $width;
+}
+
+//VARIABLES
+//rgba(167, 210, 203, 0.6);
```
Comment (haplo) [inline src/styles/mixins.scss]: Remove commented out code.

File: src/styles/mixins.scss [hunk 24]
```diff
@@ -0,0 +1,54 @@
+@import url("https://fonts.googleapis.com/css2?family=ABeeZee&family=Indie+Flower&family=Inter:wght@200;300&family=Libre+Franklin:wght@300&family=Lora:wght@600&family=Poppins:wght@200;300&family=Press+Start+2P&family=Roboto:wght@500&family=Ubuntu:wght@300&display=swap");
```
Comment (haplo) [inline src/styles/mixins.scss]: Remove all references to external fonts.

File: src/styles/photoDetail.scss [hunk 25]
```diff
@@ -0,0 +1,217 @@
+@use "./mixins.scss" as mix;
+//rgba(167, 210, 203, 0.6);
```
Comment (haplo) [inline src/styles/photoDetail.scss]: Remove commented out code.

File: src/styles/photoDetail.scss [hunk 26]
```diff
@@ -0,0 +1,217 @@
...
+    transform: translateY(0);
+    //transition: transform 0.6s ease-out; //New
+  }
+}
+.footer-container {
+  display: none;
+}
+////mobile landscape///
+@media only screen and (max-height: 575.98px) and (orientation: landscape) {
+  //enter code here
```
Comment (haplo) [inline src/styles/photoDetail.scss]: Remove comment.

File: src/styles/photoDetail.scss [hunk 27]
```diff
@@ -0,0 +1,217 @@
...
+  ##Device = Tablets, Ipads (landscape)
+  ##Screen = B/w 768px to 1024px
+*/
+
+@media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape) {
+  /* CSS */
+}
+//Laptops/////////////////////////////////////////////////////////////////
+// @media screen
+//   and (min-device-width: 1200px)
```
Comment (haplo) [inline src/styles/photoDetail.scss]: Comments don't match the media query below.

File: src/styles/photoDetail.scss [hunk 28]
```diff
@@ -0,0 +1,217 @@
...
+        padding: 0;
+        height: 4, 5vh;
+        .toggle-drawer {
+          .toggle-line {
+            display: none;
+          }
+        }
+      }
+      .info-open-drawer {
+        //  padding: 5rem; //esto mientrastanto
```
Comment (haplo) [inline src/styles/photoDetail.scss]: Remove commented out code.

File: src/utils/photoUrl.ts [hunk 29]
```diff
@@ -0,0 +1,10 @@
+function photoUrl(photoId: string, albumId?: string) {
+  // "/albums/:albumId/photos/:photoId"
```
Comment (haplo) [inline src/utils/photoUrl.ts]: Remove commented out code.

**Review by haplo — CHANGES_REQUESTED** [review]

File: src/assets/x.svg (hunk 5 above)
Comment (natalieacevedo) [reply src/assets/x.svg]: I deleted the file because I did not end up using it.

File: src/assets/hamburguer.svg (hunk 6 above)
Comment (natalieacevedo) [reply src/assets/hamburguer.svg]: I deleted the file because I did not end up using it.

File: src/components/Footer.tsx (hunk 8 above)
Comment (natalieacevedo) [reply src/components/Footer.tsx]: The three divs were result of a merge, I did not write them. I erased two and left the main footer div.

File: src/components/Photo.tsx (hunk 12 above)
Comment (natalieacevedo) [reply src/components/Photo.tsx:87]: As discussed, the image URL will be moved as an attribute of the API.

File: src/styles/mixins.scss (hunk 23 above)
Comment (natalieacevedo) [reply src/styles/mixins.scss]: Also removed google fonts since they are not going to be utilized in the project.

**Review by haplo — APPROVED** [review]
Really nice work addressing all comments! :100: 

Ready to merge! :rocket:

**Pull request merged**